python/PyQt6/core/auto_generated/qgsmaplayer.sip.in

```diff
@@ -567,13 +567,13 @@ or other problem. Child classes set this flag when initialized.
 :return: ``True`` if the layer is valid and can be accessed
 %End
 
-    QString publicSource( bool hidePassword = false ) const;
+    QString publicSource( bool redactCredentials = false ) const;
```

Comment:
this is technically an API break, because a python caller to `layer.publicSource(hidePassword=True)` will break

Reply:
I struggled with the current naming because of the double negation (what does hidePassword = false  mean : show password ... ?). Ideally we'd probably need some enumeration like RemoveCredential and HideCredential , or something like that. 

Can we take a bit of risk? Because https://github.com/search?q=publicSource%28&type=code shows a lot of hits but https://github.com/search?q=publicSource(hidePassword&type=code none and https://github.com/search?q=publicSource(+hidePassword&type=code just forks of this repository, so at least we shouldn't break any public github plugin. Especially since the addition of the hidePassword is quite recent (3.34)

Reply:
I think it's a bit of a slippery slope -- I'd rather just see a QGIS 4.0 Todo and live with the poor naming for now

Reply:
> I think it's a bit of a slippery slope -- I'd rather just see a QGIS 4.0 Todo and live with the poor naming for now

Got it. TODO added, and parameter name reverted back to original hidePassword